Use docker image tag when reporting coverage

mihails-strasuns-sociomantic opened the pull request:
Mergin code coverage bit script with matrix build changes has broken the
former because `ocean:latest` is not available anymore.

mihails-strasuns-sociomantic commented:
Will auto-merge now to fix coverage reports in other PRs, @leandro-lucarella-sociomantic please poke me if anything is wrong with it

mihails-strasuns-sociomantic commented:
Hm, @leandro-lucarella-sociomantic do you know if there is any way to run `after_script` part only once instead of per each matrix row?

mihails-strasuns-sociomantic committed:
Use docker image tag when reporting coverage

Mergin code coverage bit script with matrix build changes has broken the
former because `ocean:latest` is not available anymore.

leandro-lucarella-sociomantic commented:
> Hm, @leandro-lucarella-sociomantic do you know if there is any way to run after_script part only once instead of per each matrix row?

Nope. And I don't think it's possible, because each matrix build runs in a different VM, and AFAIK you can't pass artefacts from one VM to another (that is not using your own file server or something), so even if you could have a final step that runs only once (I think we could maybe do that with stages), then you wouldn't have the coverage files because they are not in the new VM. I think the only way to hack this is to hace some `if test "$DIST" = xenial; then codecov; fi`.

leandro-lucarella-sociomantic commented:
There seems to be some spurious network failure in one matrix job, I'm re-running it. Will merge if it's OK.

mihails-strasuns-sociomantic commented:
How about we just set `AFTER_SCRIPT=1` flag for one of matrix rows and run post-scripts from that only? D2 conversion will have the same issue.

mihails-strasuns-sociomantic committed:
Run travis-after-success script only once

This script contains various actions to be done after tests succeeds -
it is wasteful or even wrong to run it for every matrix row.

mihails-strasuns-sociomantic commented:
Updated with such change.

leandro-lucarella-sociomantic commented on ci/travis-after-success.sh:
```diff
@@ -1,6 +1,8 @@
 #!/bin/sh
 set -xe
 
+test "$AFTER_SCRIPT" = "1" || exit 0
```
Mmm, this will make the script exit with a non 0 status if the test fail, because `-e` will make the shell exit as soon as the `test` command fails, so `exit 0` is never executed. At least in theory, but it seems like tests passed, I don't know why.

leandro-lucarella-sociomantic replied:
Mmm, it seems like it's not how I remembered it. I know there is a catch in using a plain `test` when `set -e` is used, but it seems like it's not this one. It looks like the shell run the `exit 0` (which I guess it makes sense). I think the problem is only with `test ... && something`, because if the first test fails, the script stops right there.

Anyway, nevermind.

mihails-strasuns-sociomantic replied:
I will keep in mind that it is something to double-check, thanks :)
